File: java/src/main/java/com/penapereira/patterns/visitor/ConcreteElementB.java
package com.penapereira.patterns.visitor;

/** Element B; accepts a visitor and exposes operationB to it. */
public final class ConcreteElementB implements Element {

    @Override
    public void accept(Visitor visitor) {
        visitor.visitConcreteElementB(this);
    }

    public String operationB() {
        return "B";
    }
}

File: typescript/src/visitor/concrete-element-b.ts
import type { Element } from "./element.ts";
import type { Visitor } from "./visitor.ts";

export class ConcreteElementB implements Element {
  accept(visitor: Visitor): void {
    visitor.visitConcreteElementB(this);
  }

  operationB(): string {
    return "B";
  }
}
